test/functional/test_framework/mininode.py

```diff
@@ -79,18 +80,38 @@ def __init__(self):
     def is_connected(self):
         return self._transport is not None
 
-    def peer_connect(self, dstaddr, dstport, net="regtest"):
+    def peer_connect(self, dstaddr, dstport, net="regtest", node_outgoing=False):
         assert not self.is_connected
         self.dstaddr = dstaddr
         self.dstport = dstport
         # The initial message to send after the connection was made:
         self.on_connection_send_msg = None
         self.recvbuf = b""
         self.network = net
-        logger.debug('Connecting to Bitcoin Node: %s:%d' % (self.dstaddr, self.dstport))
+        self.node_outgoing = node_outgoing
 
         loop = NetworkThread.network_event_loop
-        conn_gen_unsafe = loop.create_connection(lambda: self, host=self.dstaddr, port=self.dstport)
+
+        if self.node_outgoing:
+            logger.debug('Connecting from Bitcoin Node: %s:%d' % (self.dstaddr, self.dstport))
+
+            listen_sock = socket.socket()
+            listen_sock.bind(('127.0.0.1', 0))
+            listen_sock.listen(1)
+            listen_port = listen_sock.getsockname()[1]
+            self.rpc.addnode('127.0.0.1:%u' % (listen_port,), 'onetry', False)
+            (sock, addr) = listen_sock.accept()
```

Comment:
Use `(sock, _) = listen_sock.accept()` to clarify that the second element in the tuple is discarded intentionally